sdk/storage/azure-storage-stress/src/main/java/com/azure/storage/stress/FaultInjectingHttpPolicy.java

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.storage.stress;
+
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpPipelineNextSyncPolicy;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import com.azure.core.util.UrlBuilder;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+import reactor.util.function.Tuple2;
+import reactor.util.function.Tuples;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.azure.core.http.HttpHeaderName.X_MS_CLIENT_REQUEST_ID;
+
+public class FaultInjectingHttpPolicy implements HttpPipelinePolicy {
+    private static final ClientLogger LOGGER = new ClientLogger(FaultInjectingHttpPolicy.class);
+    private static final HttpHeaderName UPSTREAM_URI_HEADER = HttpHeaderName.fromString("X-Upstream-Base-Uri");
+    private static final HttpHeaderName HTTP_FAULT_INJECTOR_RESPONSE_HEADER = HttpHeaderName.fromString("x-ms-faultinjector-response-option");
+    private static final HttpHeaderName SERVER_REQUEST_ID_HEADER = HttpHeaderName.fromString("x-ms-request-id");
+    private final boolean https;
+    private final boolean isRequestFaulted;
+
+
+    private final List<Tuple2<Double, String>> probabilities;
+
+    public FaultInjectingHttpPolicy(boolean https, FaultInjectionProbabilities probabilities, boolean isRequestFaulted) {
+        this.https = https;
+        // If this is true, requests will be faulted. Else responses will be faulted.
+        this.isRequestFaulted = isRequestFaulted;
+
+        this.probabilities = new ArrayList<>();
+
+        if (this.isRequestFaulted) {
```

Comment:
Might I recommend removing `probabilities` and just have two internal methods that return the specific handling based on whether `isRequestFaulted` is true? It'll make `faultInjectorHandling` more verbose but it'll be easier to understand by looking at it quickly.